### PR TITLE
Fix invalid credential error handling in authentication process

### PR DESCRIPTION
**Pull request by Didainius**
This PR builds on top of https://github.com/vmware/go-vcloud-director/pull/536 and only adds additional tests as the real fix is in go-vcloud-director SDK.

We had an unhandled HTTP 401 (Unauthorized) error in the password authentication flow and as a result it would not return an error when user supplies invalid credentials. The errors then become very confusing and the user is left helpless to do his guesswork.

**Before**, this would cause very confusing errors similar to:
```
Error: could not find any extended Provider VDC with name ext-net-name: error retrieving query: API Error: 406: The request has invalid accept header: Invalid API version requested. Supported API versions are: [37.0.0-alpha, 36.3, 36.2, 36.1, 36.0, 35.2, 35.0, 34.0, 33.0 [D], 32.0 [D], 31.0 [D]] ([D] indicates deprecated versions)
│
│   with data.vcd_provider_vdc.nsxt-pvdc,
│   on Org-creation-Main.tf line 47, in data "vcd_provider_vdc" "nsxt-pvdc":
│   47: data "vcd_provider_vdc" "nsxt-pvdc" {
│
╵
╷
│ Error: could not find external network V2 by name 'ext-net-name': could not find external network by name: error getting all pages for endpoint https://HOSTNAME/cloudapi/1.0.0/externalNetworks/: error in HTTP GET request: NOT_ACCEPTABLE - The request has invalid accept header: Invalid API version requested. Supported API versions are: [37.0.0-alpha, 36.3, 36.2, 36.1, 36.0, 35.2, 35.0, 34.0, 33.0 [D], 32.0 [D], 31.0 [D]] ([D] indicates deprecated versions)
│
│   with data.vcd_external_network_v2.nsxt-ext-net,
│   on Org-creation-Main.tf line 98, in data "vcd_external_network_v2" "nsxt-ext-net":
│   98: data "vcd_external_network_v2" "nsxt-ext-net" {
```

**After** the fix, the user would get similar error which directly hints that one should check his credentials:
```
   auth_test.go:310: Step 1/1 error: Error running pre-apply refresh: exit status 1

        Error: something went wrong during authentication: error authorizing: received response HTTP 401 (Unauthorized). Please check if your credentials are valid

          with provider["registry.terraform.io/hashicorp/vcd"],
          on terraform_plugin_test.tf line 2, in provider "vcd":
           2:                   provider "vcd" {
```

Three additional subtests are added to `TestAccAuth` that expect an error:
* `TestAccAuth/InvalidSystemUserAndPasswordWithDefaultOrgAndVdc`
* `TestAccAuth/InvalidPassword,AuthType=integrated`
* `TestAccAuth/InvalidSystemUserAndPassword,AuthType=integrated`

### CODE REVIEW

**Review by adambarreiro — APPROVED** [review]
LGTM!

File: .changes/v3.9.0/962-bug-fixes.md [hunk 1]
```diff
@@ -0,0 +1 @@
+* Fix a bug, that would not return an error while authenticating with invalid password [GH-962]
```
Comment (lvirbalas) [inline .changes/v3.9.0/962-bug-fixes.md:1]: Actually, I like your other PR's in govcd description better 
https://github.com/vmware/go-vcloud-director/pull/536/files#diff-2cb2af9c1a524452ed1490c48b24cf528ea3e20c1a2785bba2cc6533635f1168R1

```suggestion
* Fix a bug that prevented returning a specific error while authenticating provider with invalid password [GH-962]
```

**Review by lvirbalas — APPROVED** [review]
Just one suggestion for changelog.

File: .changes/v3.9.0/962-bug-fixes.md (hunk 1 above)
Comment (Didainius) [reply .changes/v3.9.0/962-bug-fixes.md]: Changed.

**Pull request merged**